tikv: add doc for new config `max-apply-unpersisted-log-limit` (#17857)
pingcap/docs: tikv-configuration-file.md

+ Default value: `"30s"`
+ Minimum value: `0`

> ### `max-apply-unpersisted-log-limit` <span class="version-mark">New in v8.2.0</span>

Author: ```suggestion
### `max-apply-unpersisted-log-limit` <span class="version-mark">New in v8.1.0</span>
```

> + The maximum number of committed but not persisted Raft logs that are allowed to be applied.

Reviewer: ```suggestion
+ The maximum number of committed but not persisted Raft logs that can be applied.
```

+ Default value: `4096`
+ Minimum value: `0`

### `max-apply-unpersisted-log-limit` <span class="version-mark">New in v8.2.0</span>

+ The maximum number of committed but not persisted Raft logs that are allowed to be applied.

    + Setting this configuration item to a value greater than `0` enables the TiKV node to apply committed but not persisted Raft logs in advance, effectively reducing long-tail latency caused by IO jitter on that node. However, it might also increase the memory usage of TiKV and the disk space occupied by Raft logs.

>     + Setting this configuration item to `0` disables this feature, meaning TiKV must wait for the Raft logs to be both committed and persisted before applying them. This is consistent with the behavior before v8.2.0.

Author: ```suggestion
    + Setting this configuration item to `0` disables this feature,   indicating that that TiKV must wait until Raft logs are both committed and persisted before applying them. This behavior is consistent with the behavior before v8.2.0.
```
